### PR TITLE
Refactor real-time RC to use tasks for auth token requests

### PR DESCRIPTION
**Pull request by qdpham13**
Update real-time http methods to use Tasks for auth tokens and FID (future).

### CODE REVIEW

File: firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java [hunk 1]
```diff
@@ -315,13 +304,37 @@ private URL getUrl() {
...
+    Task<InstallationTokenResult> installationAuthTokenTask = firebaseInstallations.getToken(false);
+    // When the auth token task has finished, set up HTTP connection with headers and params.
+    return Tasks.whenAllComplete(installationAuthTokenTask)
+        .continueWithTask(
+            this.scheduledExecutorService,
+            (completedInstallationTask) -> {
+              if (!installationAuthTokenTask.isSuccessful()) {
+                return Tasks.forException(
+                    new FirebaseRemoteConfigClientException(
+                        "Firebase Installations failed to get installation ID for real-time.",
```
Comment (danasilver) [inline firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java:318]: I think the exception should be `installation auth token` here

File: firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java [hunk 2]
```diff
@@ -455,59 +468,83 @@ public void beginRealtimeHttpStream() {
...
+            this.scheduledExecutorService,
+            (completedHttpUrlConnectionTask) -> {
+              Integer responseCode = null;
+              HttpURLConnection httpURLConnection = null;
+              setIsHttpConnectionRunning(true);
+
+              try {
+                // If HTTP connection task failed throw exception to move to the catch block.
+                if (!httpURLConnectionTask.isSuccessful()) {
+                  throw new IOException(httpURLConnectionTask.getException());
```
Comment (danasilver) [inline firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java:485]: Thinking: More refactoring would propagate the failed task, but we're limiting the scope of this change. It's a bit confusing to throw the `IOException` here just to handle it a few lines below.

File: firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java [hunk 3]
```diff
@@ -455,59 +468,83 @@ public void beginRealtimeHttpStream() {
...
+    // Make async call to create and setup HTTP connection.
+    Task<HttpURLConnection> httpURLConnectionTask = createRealtimeConnection();
+    // When the connection task has finished, begin real-time actions.
+    Tasks.whenAllComplete(httpURLConnectionTask)
+        .continueWith(
+            this.scheduledExecutorService,
+            (completedHttpUrlConnectionTask) -> {
+              Integer responseCode = null;
+              HttpURLConnection httpURLConnection = null;
+              setIsHttpConnectionRunning(true);
```
Comment (danasilver) [inline firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java:480]: Should this call happen after checking that the task completed successfully?

**Review by danasilver — APPROVED** [review]
Couple thoughts and one comment/question.

**Review by danasilver — APPROVED** [review]

File: firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java (hunk 2 above)
Comment (qdpham13) [reply firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java:484]: Yeah I think more refactoring can be done to handle the failed task better, I threw the `IOException` here b/c I wanted to keep the same try/catch structure that we had here and to reduce the scope of these changes

File: firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java (hunk 1 above)
Comment (qdpham13) [reply firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java]: updated

**Pull request merged**